nestkernel/event_delivery_manager.cpp

```diff
@@ -426,12 +415,13 @@ EventDeliveryManager::gather_spike_data_( std::vector< SpikeDataT >& send_buffer
     // as all chunk-end entries, have marker DEFAULT.
     set_end_marker_( send_buffer_position, send_buffer, local_max_spikes_per_rank );
 
-#ifdef TIMER_DETAILED
-    {
-      sw_collocate_spike_data_.stop();
-      sw_communicate_spike_data_.start();
-    }
+    sw_collocate_spike_data_.stop();
+#if defined( HAVE_MPI ) && defined( TIMER_DETAILED )
+    kernel().get_mpi_synchronization_stopwatch().start();
+    kernel().mpi_manager.synchronize();
+    kernel().get_mpi_synchronization_stopwatch().stop();
 #endif
+    sw_communicate_spike_data_.start();
```

Comment:
So far, we always considered the synchronization as a part of communication. Communication = synchronization + data exchange. I think we should keep it this way and have the communicate_spike_data Stopwatch start before synchronization.